### PR TITLE
[MU3 Backend] ENG-59: Fix missing secondary beam breaks

### PR DESCRIPTION
**Pull request by iveshenry18**
Resolves: [ENG-59](https://mu--se.atlassian.net/jira/software/projects/ENG/boards/21?selectedIssue=ENG-59): Fix missing breaks in secondary beams

Previously, only the first beam was taken into account when importing
Beam::Mode. This commit adds a mechanism to collect the beam type
for all beams and make a more correct choice about Beam::Mode (including
BEGIN32 and BEGIN64). Additionally, it adds handling of these modes to
the export code. Now the round trip from .mscx -> .xml -> .mscx is
correct and complete *when beam modes are specified explicitly in the
.mscx*; however, more thorough handling of Beam::Mode::AUTO (where it
results in the equivalent of BEGIN32 or BEGIN64) on export needs to be
done.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [x] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Review by oktophonie — APPROVED** [review]

**Review by vpereverzev — APPROVED** [review]

File: importexport/musicxml/exportxml.cpp [hunk 1]
```diff
@@ -2923,30 +2923,43 @@ static void writeBeam(XmlWriter& xml, ChordRest* const cr, Beam* const b)
...
             }
       // find beam type and write
       for (int i = 1; i <= blc; ++i) {
             QString text;
-            if (blp < i && bln >= i) text = "begin";
+            // TODO: correctly handle Beam::Mode::AUTO
+            // when equivalent to BEGIN32 or BEGIN64
+            if ((blp < i && bln >= i)
+             || bmc == Beam::Mode::BEGIN32 && i > 1
+             || bmc == Beam::Mode::BEGIN64 && i > 2)
```
Comment (Jojo-Schmitz) [inline importexport/musicxml/exportxml.cpp:2954]: this introduces compiler warning, suggetion to add parens or the `&&` conditions

File: importexport/musicxml/exportxml.cpp [hunk 2]
```diff
@@ -2923,30 +2923,43 @@ static void writeBeam(XmlWriter& xml, ChordRest* const cr, Beam* const b)
...
+                  text = "begin";
             else if (blp < i && bln < i) {
                   if (bln > 0) text = "forward hook";
                   else if (blp > 0) text = "backward hook";
                   }
-            else if (blp >= i && bln < i)
+            else if ((blp >= i && bln < i)
+                  || bmn == Beam::Mode::BEGIN32 && i > 1
+                  || bmn == Beam::Mode::BEGIN64 && i > 2)
```
Comment (Jojo-Schmitz) [inline importexport/musicxml/exportxml.cpp:2962]: same

File: importexport/musicxml/exportxml.cpp (hunk 1 above)
Comment (iveshenry18) [reply importexport/musicxml/exportxml.cpp:2954]: #8554

Comment (Jojo-Schmitz) [reply importexport/musicxml/exportxml.cpp:2954]: Thanks!

Comment (iveshenry18) [reply importexport/musicxml/exportxml.cpp:2954]: Likewise! I've been trying to keep a closer eye on warnings as I compile locally; hopefully no more will slip through!

**Pull request merged**